tests/appsec/test_automated_login_events.py

```diff
@@ -1348,3 +1218,703 @@ def validate_iden(meta):
+            if context.library not in libs_without_user_id:
+                assert meta["appsec.events.users.signup.usr.id"] == NEW_USER_HASH
+                assert meta["_dd.appsec.usr.id"] == NEW_USER_HASH
+
+
+DISABLED = ("datadog/2/ASM_FEATURES/auto-user-instrum/config", {"auto_user_instrum": {"mode": "disabled"}})
+IDENTIFICATION = ("datadog/2/ASM_FEATURES/auto-user-instrum/config", {"auto_user_instrum": {"mode": "identification"}})
+ANONYMIZATION = ("datadog/2/ASM_FEATURES/auto-user-instrum/config", {"auto_user_instrum": {"mode": "anonymization"}})
+
+
+@rfc("https://docs.google.com/document/d/1RT38U6dTTcB-8muiYV4-aVDCsT_XrliyakjtAPyjUpw")
+@features.user_monitoring
+@scenarios.appsec_auto_events_rc
+class Test_V3_Login_Events_RC:
+    def _send_rc_and_execute_request(self, config):
+        config_state = rc.rc_state.set_config(*config).apply()
+        request = weblog.post("/login?auth=local", data=login_data(context, USER, PASSWORD))
+        return {"config_state": config_state, "request": request}
+
+    def _assert_response(self, test, validation):
+        config_state, request = test["config_state"], test["request"]
+
+        assert config_state[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert request.status_code == 200
+
+        spans = [s for _, _, s in interfaces.library.get_spans(request=request)]
+        assert spans, "No spans to validate"
+        for span in spans:
+            meta = span.get("meta", {})
+            validation(meta)
+
+    def setup_rc(self):
+        self.disabled = self._send_rc_and_execute_request(DISABLED)
+        self.identification = self._send_rc_and_execute_request(IDENTIFICATION)
+        self.anonymization = self._send_rc_and_execute_request(ANONYMIZATION)
+
+    def test_rc(self):
+        def validate_disabled(meta):
+            assert "_dd.appsec.events.users.login.success.auto.mode" not in meta
+
+        def validate_anon(meta):
+            assert meta["_dd.appsec.events.users.login.success.auto.mode"] == "anonymization"
+
+        def validate_iden(meta):
+            assert meta["_dd.appsec.events.users.login.success.auto.mode"] == "identification"
+
+        self._assert_response(self.disabled, validate_disabled)
+        self._assert_response(self.identification, validate_iden)
+        self._assert_response(self.anonymization, validate_anon)
+
+
+CONFIG_ENABLED = (
+    "datadog/2/ASM_FEATURES/asm_features_activation/config",
+    {"asm": {"enabled": True}},
+)
+
+BLOCK_USER = (
+    "datadog/2/ASM_DD/rules/config",
+    {
+        "version": "2.1",
+        "metadata": {"rules_version": "1.2.6"},
+        "rules": [
+            {
+                "id": "block-users",
+                "name": "Block User Addresses",
+                "tags": {"type": "block_user", "category": "security_response"},
+                "conditions": [
+                    {
+                        "parameters": {"inputs": [{"address": "usr.login"}], "data": "blocked_users"},
```

Comment:
Here you're either using the wrong address, or the wrong data. `social-security-id` is the `usr.id`, not the `usr.login`. I'm assuming you want to test it's blocking on `usr.id` ?

Reply:
fixed in 5894341d6f4d694c4f8970a948d1f9b92a473b70